### PR TITLE
fix: check prometheus init status before use in batch processor (#8065)

### LINKED ISSUE #8065: bug: batch_process:push method failed in 2.15.0
### Current Behavior

Method "batch_process:push" in batch-processor.lua report error: attempt to index a nil value.
` if prometheus and not batch_metrics and self.name
       and self.route_id and self.server_addr then
        batch_metrics = prometheus.get_prometheus():gauge("batch_process_entries",
                                                          "batch process remaining entries",
                                                          {"name", "route_id", "server_addr"})
    end`
the code determine prometheus but not  prometheus.get_prometheus() .

### Expected Behavior

In the version 2.12, the code is different and it works.
`if not batch_metrics and prometheus.get_prometheus() and self.name
       and self.route_id and self.server_addr then
        batch_metrics = prometheus.get_prometheus():gauge("batch_process_entries",
                                                          "batch process remaining entries",
                                                          {"name", "route_id", "server_addr"})
    end`
The code use "if not batch_metrics and **prometheus.get_prometheus()** ", so it works well.

### Error Logs

2022/10/10 19:07:52 [error] 15074#0: *1109498 failed to run log_by_lua*: .../software/apisix-2.15.0/apisix/utils/batch-processor.lua:189: attempt to index a nil value
stack traceback:
        .../software/apisix-2.15.0/apisix/utils/batch-processor.lua:189: in function 'push'
        ...e/apisix-2.15.0/apisix/utils/batch-processor-manager.lua:121: in function 'add_entry_to_new_processor'

### Steps to Reproduce

1.Run apisix in docker without promethus.
2.config a route with kafka-logger
3.fire a request, get the error.

### Environment

- APISIX version 2.15 (run `apisix version`):
- Operating system linux (run `uname -a`):

### PR DESCRIPTION
**Pull request by monkeyDluffy6017**
### Description

check prometheus init status before use in batch processor 

Fixes #8065 

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [x] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [x] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

**Review by tzssangglass — DISMISSED** [review]

**Comment by tzssangglass** [comment]
If the error of #8065 cannot be reproduced, then the fix does not work.

File: t/plugin/kafka-logger.t [hunk 1]
```diff
@@ -708,3 +708,52 @@ hello world
...
+            end
+            ngx.say(body)
+        }
+    }
+--- response_body
+passed
+
+
+
+=== TEST 25: access
```
Comment (spacewander) [inline t/plugin/kafka-logger.t:754]: The test can pass even if I remove the change in the apisix/utils/batch-processor.lua.

Comment (HT404) [reply t/plugin/kafka-logger.t:754]: Can you show us the route information you configured?

Comment (soulbird) [reply t/plugin/kafka-logger.t:754]: It means to delete your change, the newly added test case can also pass.

Comment (monkeyDluffy6017) [reply t/plugin/kafka-logger.t:754]: I can not reproduce the problem, but the test case can reach the code line where the user report, the test case is not a good   one, i will try to reproduce again.

Comment (monkeyDluffy6017) [reply t/plugin/kafka-logger.t:754]: I have reproduce the problem,  request the router with `prometheus` plugin disabled will cause the problem

**Review by spacewander — APPROVED** [review]

**Review by tzssangglass — APPROVED** [review]

**Review by soulbird — APPROVED** [review]

**Pull request merged**